src/utils/metadata.ts

```diff
@@ -176,3 +177,21 @@ export function fetchIdpsMetadata(
       return mapIpdMetadata(idpMetadata, idpIds);
     });
 }
+
+export function parseStartupSpidStrategy(
+  idpsMetadata: Record<string, string>
+): Record<string, IDPEntityDescriptor> {
+  return mapIpdMetadata(
+    Object.values(idpsMetadata).reduce(
+      (prev, metadataXML) => [
+        ...prev,
+        ...parseIdpMetadata(metadataXML).fold(
+          () => [],
+          _ => _
+        )
+      ],
+      [] as ReadonlyArray<IDPEntityDescriptor>
```

Comment:
you may want to try using `StrMap(idpsMetadata).reduce` instead of reduce on the record (StrMap version is typed).

Reply:
Fixed in a4bdb13
Much better solution :)